url6f0c51b1b4_TheFakeMontyOnTheRun_open688_tgz-pJ8-br_odb_open688_app_Open688ServerJ8/src/br/odb/open688/app/Station.java

```diff
@@ -0,0 +1,50 @@
+package br.odb.open688.app;
+
+import java.io.IOException;
+
+import br.odb.gameapp.ConsoleApplication;
+import br.odb.open688.app.net.TelnetClientServer;
+
+public class Station extends TelnetClientServer implements Runnable {
+
+	private Open688Server app;
+
+	public Station(int clientId ) {
+		super( clientId );
+	}
+
+	public void setMasterClient(ConsoleApplication app) {
+
+		this.app = (Open688Server) app;
+
+		new Thread(this).start();
+	}
+
+	@Override
+	public String getInput(String msg) {
+		return super.getInput(msg);
+	}
+
+	@Override
+	public void run() {
+
+		openServer();
+
+		while (true) {
+			try {
+				Thread.sleep(100);
+			} catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)